plugins/ksr/plugin_impl_ksr.go

```diff
@@ -201,13 +208,75 @@ func (plugin *Plugin) AfterInit() error {
 	plugin.policyReflector.Start()
 	plugin.serviceReflector.Start()
 	plugin.endpointsReflector.Start()
+
+	go plugin.monitorEtcdStatus(plugin.stopCh)
+
 	return nil
 }
 
 // Close stops all reflectors.
 func (plugin *Plugin) Close() error {
 	close(plugin.stopCh)
-	safeclose.CloseAll(plugin.nsReflector, plugin.podReflector, plugin.policyReflector)
+	safeclose.CloseAll(plugin.nsReflector, plugin.podReflector, plugin.policyReflector,
+		plugin.serviceReflector, plugin.endpointsReflector)
 	plugin.wg.Wait()
 	return nil
 }
+
+// stopDataStoreUpdates triggers all Reflectors to stop updating the Etcd
+// data store.
+func (plugin *Plugin) stopDataStoreUpdates() {
+	plugin.Log.Info("stopDataStoreUpdates")
+
+	plugin.nsReflector.StopDataStoreUpdates()
+	plugin.podReflector.StopDataStoreUpdates()
+	plugin.policyReflector.StopDataStoreUpdates()
+	plugin.serviceReflector.StopDataStoreUpdates()
+	plugin.endpointsReflector.StopDataStoreUpdates()
+}
+
+// syncDataStoreWithK8sCache triggers all Reflectors to start the
+// reconciliation their respective K8s caches with the data store.
+func (plugin *Plugin) reconcileDataStoreWithK8sCache() {
+	plugin.Log.Info("syncDataStoreWithK8sCache")
+
+	plugin.nsReflector.SyncDataStoreWithK8sCache()
+	plugin.podReflector.SyncDataStoreWithK8sCache()
+	plugin.policyReflector.SyncDataStoreWithK8sCache()
+	plugin.serviceReflector.SyncDataStoreWithK8sCache()
+	plugin.endpointsReflector.SyncDataStoreWithK8sCache()
+}
+
+// monitorEtcdStatus monitors the KSR's connection to the Etcd Data Store.
+func (plugin *Plugin) monitorEtcdStatus(closeCh chan struct{}) {
+	for {
+		select {
+		case <-closeCh:
+			plugin.Log.Info("Closing")
+			return
+		case <-time.After(1 * time.Second):
+			sts := plugin.StatusMonitor.GetAllPluginStatus()
+			for k, v := range sts {
+				if k != "etcdv3" {
+					return
```

Comment:
I think would be better to use either `continue` instead of `return` or map access `sts["etcdv3"]`. Otherwise, if AllPluginStatus contains status of a plugin different from etcdv3, monitoring is stopped.